xoa/data_source_hosts.go

```diff
@@ -0,0 +1,86 @@
+package xoa
+
+import (
+	"github.com/ddelnano/terraform-provider-xenorchestra/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+)
+
+func dataSourceXoaHosts() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHostsRead,
+		Schema: map[string]*schema.Schema{
+			"master": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hosts": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+			},
+			"pool": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"sort_by": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"sort_order": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceHostsRead(d *schema.ResourceData, m interface{}) error {
+	c := m.(*client.Client)
+	poolLabel := d.Get("pool").(string)
+	tags := d.Get("tags").([]interface{})
+
+	err := d.Set("pool", poolLabel)
```

Comment:
In data sources you don't need to call `Set("attr", ...)` on a value that was passed in as an input. Notice how the `host` data source does not do that [here](https://github.com/terra-farm/terraform-provider-xenorchestra/blob/6d0b85c152094efd9ee16ad0c8b6a080b701c305/xoa/data_source_host.go#L22-L43).

Terraform will treat the argument value as the resulting state for that attribute.